lib/ansible/modules/cloud/amazon/redshift_iam_roles.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+module: redshift_iam_roles
+short_description: This module adds/removes IAM roles to/from Redshift clusters.
+version_added: 2.6
+author: Aaron Smith (@slapula)
+description:
+    - "This module adds or removes a list of IAM roles to a give Redshift Cluster"
+options:
+    cluster:
+        description:
+            - "This is the cluster you'd like to modify"
+        required: true
+    state:
+        description:
+            - "Desired state of the list of IAM roles on the give Redshift cluster"
+        required: true
+        choices: ['present', 'absent']
+    roles:
+        description:
+            - "This is a list of IAM roles (Full ARN) to add to the Redshift cluster. Roles attached to the cluster that are not on the list will be removed."
+        required: true
+extends_documentation_fragment:
+    - ec2
+    - aws
+'''
+
+EXAMPLES = '''
+- name: add IAM roles to a redshift cluster
+  redshift_iam_roles:
+    cluster: "staging"
+    state: present
+    roles:
+        - "arn:aws:iam::123456789012:role/superuser"
+        - "arn:aws:iam::123456789012:role/new_hotness"
+
+- name: remove all IAM roles from a redshift cluster
+  redshift_iam_roles:
+    cluster: "staging"
+    state: absent
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info, camel_dict_to_snake_dict, HAS_BOTO3
+import traceback
+
+try:
+    from botocore.exceptions import ClientError, BotoCoreError
+except ImportError:
+    pass  # will be detected by imported HAS_BOTO3
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            cluster=dict(type='str', required=True),
+            state=dict(type='str', required=True, choices=['present', 'absent']),
+            roles=dict(type='list', required=True)
+        )
+    )
+
+    module = AnsibleModule(
```

Comment:
If you update this to use AnsibleAWSModule you will be able to simplify quite a few things, like exception handling and boto3 connection creation.